lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -261,6 +264,42 @@ sub _get_dirty_status {
     return "$status on $when";
 }
 
+# Which repo on OBS should trigger sync
+sub _get_api_repo {
+    my ($c, $alias) = @_;
+    my $helper = $c->obs_rsync;
+    # doesn't depend on batch, so just strip it out
+    my ($project, undef) = $helper->split_alias($alias);
+    my $home = $helper->home;
+    my $ret  = _get_first_line(Mojo::File->new($home, $project, $api_repo_filename));
+    return $ret if $ret;
+    return "images";
+}
+
+# Which package on OBS should be checked for being published on obs
+sub _get_api_package {
+    my ($c, $alias) = @_;
+    my $helper = $c->obs_rsync;
+    # doesn't depend on batch, so just strip it out
+    my ($project, undef) = $helper->split_alias($alias);
+    my $home             = $helper->home;
+    my $api_package_file = Mojo::File->new($home, $project, $api_package_filename);
+    return '000product' unless -f $api_package_file;
+    return _get_first_line($api_package_file);
+}
+
+# Build url to check dirty status for project
+sub _get_api_dirty_status_url {
+    my ($c, $project) = @_;
+    my $helper = $c->obs_rsync;
+    my $url    = $helper->project_status_url;
+    return "" unless $url;
+    $url =~ s/%%PROJECT/$project/g;
+    my $package = $helper->get_api_package($project);
+    $url = "$url?package=$package" if $package;
```

Comment:
```suggestion
    $url->query({package => $package}) if $package;
```

Reply:
Url is a string, so it doesn't work this way

Reply:
Yes, the suggested code is wrong. But the point is that you should make the URL with `Mojo::URL` so parameters are correctly encoded.